packages/docs/docs/advanced/custom-components.mdx

```diff
@@ -0,0 +1,141 @@
+---
+slug: /custom-components
+---
+
+import AnimationPlayer from '@site/src/components/AnimationPlayer';
+import CodeBlock from '@theme/CodeBlock';
+import source from '!!raw-loader!@motion-canvas/examples/src/scenes/components-switch';
+
+# Components
+
+<AnimationPlayer name="components" banner />
+
+Components are reusable parts of code such as `Rect` and `Circle` which can be
+ scenes. To use a component in a scene, add it to the view and
+provide arguments to the component.
+
+```tsx
+<Switch
+    initialState={false}
+/>
+```
+To define what arguments a component will take in, first define an interface. All 
+properties of the interface must be wrapped in [SignalValue<>](/api/core/signals#SignalValue) 
+as such:
+```tsx
+//  You can extend an existing props interface
+//  such as LayoutProps, ShapeProps or NodeProps to
+//  include their properties alongside the ones you 
+//  define
+
+export interface SwitchProps extends NodeProps {
+    initialState?: SignalValue<boolean>
+
+    // We dont use color here because we want
+    // to be able to pass hex strings and rgb
+    // values to accent rather than a `Color`
+    accent?: SignalValue<PossibleColor>
+}
+```
+Next, create a component class for your components. The component class must extend [Node](/api/2d/components/Node)
+or one of its subclasses. If you dont want to inherit any methods from an existing component, extend your class from
+`Node`. It is advisable to extend from a component most similar to the custom component you are building. For instance, 
+if you were to make a component including a [Layout](/api/2d/components/Layout), you should extend [Layout](/api/2d/components/Layout)
+and [LayoutProps](/api/2d/components/LayoutProps).
+```tsx
+export interface SwitchProps extends NodeProps { ... }
+
+export class Switch extends Node { ... }
+```
+
+To use the properties defined in the interface, your class _must_ contain a property with the
+same name. Motion Canvas provides type decorators to facilitate this like `@initial()` and
+`@signal()`. Click [here](/docs/signals) for more information on signals.
+
+Here is an example of how you would define such properties:
+```tsx
+export class Switch extends Node {
+    // @initial - sets the property to an
+    // initial value if it was not provided.
+    // The @initial decorator is optional.
```

Comment:
Since this is on a comment on the `@initial` decorator, it's probably fine to say "This decorator is optional," or even update the beginning of this comment to "@initial - optional, sets the property to an..."

Reply:
ill go with the latter